File: catalog-browser/src/entities/Product.ts
export interface Product {
  id: string
  name: string,
  price: number,
  description: string,
  imageSrc: string,
  imageAlt: string
}

export class ProductModel implements Product{
  public id: string = '';

  constructor(public name: string,
              public price: number,
              public description: string,
              public imageSrc: string,
              public imageAlt: string) {
  }
}

File: src/main/java/mart/mono/purchases/Product.java
package mart.mono.purchases;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.xml.catalog.Catalog;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Product {

    private UUID id;

    private Catalog catalog;

    private String name;
    private String price;
    private String description;
    private String imageSrc;
    private String imageAlt;

    private int quantity;
}
